mira/sources/acsets/stockflow.py

```diff
@@ -54,20 +88,26 @@ def template_model_from_stockflow_ascet_json(model_json) -> TemplateModel:
     templates = []
 
     for flow in flows:
-        # First identify parameters and stocks in the flow expression
-        params_in_expr = re.findall(r"p\.([^()*+-/ ]+)", flow["ϕf"])
-        stocks_in_expr = re.findall(r"u\.([^()*+-/ ]+)", flow["ϕf"])
-        # We can now remove the prefixes from the expression
-        expression_str = flow["ϕf"].replace("p.", "").replace("u.", "")
-
-        # Turn each str symbol into a sympy.Symbol and add to dict of symbols
-        # if not present before and also turn it into a Parameter object to be
-        # added to tm
-        for str_symbol in set(params_in_expr + stocks_in_expr):
-            if symbols.get(str_symbol) is None:
+        expression_str = flow["ϕf"]
+
+        # identify all operands in expression
+        flow_operands = re.findall(r"\b\w[\w.]*\b", expression_str)
```

Comment:
The alternative here would be to safe parse the expression first, then get the free_symbols attribute to get all the stock names and parameter names.